apps/web/components/apps/installation/ConfigureStepCard.tsx

```diff
@@ -1,61 +1,158 @@
+      setAppData={getAppDataSetter(slug as EventTypeAppsList, categories, credentialId)}
+    />
+  );
+};
+
+const EventTypeAppSettingsForm = forwardRef<HTMLButtonElement, EventTypeAppSettingsWrapperProps>(
+  function EventTypeAppSettingsForm(props, ref) {
+    const { handleDelete, onSubmit, eventType } = props;
+
+    const formMethods = useForm<TFormType>({
+      defaultValues: {
+        metadata: eventType?.metadata,
+      },
+    });
+
+    return (
+      <Form
+        form={formMethods}
+        id={`eventtype-${eventType.id}`}
+        handleSubmit={() => {
+          const data = formMethods.getValues("metadata");
+          onSubmit(data);
+        }}>
+        <div>
+          <div className="sm:border-subtle bg-default relative border p-4 sm:rounded-md dark:bg-black">
+            <div>
+              <span className="text-default font-semibold ltr:mr-1 rtl:ml-1">{eventType.title}</span>{" "}
+              <small className="text-subtle hidden font-normal sm:inline">
+                /{eventType.team ? eventType.team.slug : props.userName}/{eventType.slug}
+              </small>
+            </div>
+            <EventTypeAppSettingsWrapper {...props} />
+            <X className="absolute right-4 top-4 h-4 w-4 cursor-pointer" onClick={() => handleDelete()} />
+            <button type="submit" ref={ref}>
+              Save
+            </button>
+          </div>
+        </div>
+      </Form>
+    );
+  }
+);
+
+export const ConfigureStepCard: FC<ConfigureStepCardProps> = ({
+  loading,
+  selectedEventTypeIds,
+  ...props
+}) => {
+  const { control } = useFormContext<TEventTypesForm>();
+  const { fields, update } = useFieldArray({
+    control,
+    name: "eventTypes",
+    keyName: "fieldId",
+  });
+  const submitRefs = useRef<Array<React.RefObject<HTMLButtonElement>>>([]);
+  submitRefs.current = selectedEventTypeIds.map(
+    (_ref, index) => (submitRefs.current[index] = React.createRef<HTMLButtonElement>())
+  );
+
+  return (
+    <div className="mt-8">
+      <div className="flex flex-col space-y-6">
+        {fields.map((field, index) => {
+          return (
+            field.selected && (
+              <EventTypeAppSettingsForm
+                key={field.fieldId}
+                eventType={field}
+                handleDelete={() => {
+                  update(index, { ...field, selected: false });
+                }}
+                onSubmit={(data) => {
+                  console.log("ddatadatadatadataata: ", data);
+                  update(index, { ...field, metadata: data });
+                }}
+                ref={submitRefs.current[index]}
+                {...props}
+              />
+            )
+          );
+        })}
+      </div>
+      <Button
+        className="text-md mt-6 w-full justify-center"
+        onClick={() => {
+          // submitRefs.current[0].current?.click();
+          submitRefs.current.map((ref) => ref.current?.click());
```

Comment:
@joeauyeung 
is there a better way to handle this?
its getting kinda messy?


the flow so far - 
[LOOM](https://www.loom.com/share/d6e517832a5c47cf8fdf313ba165e0c1?sid=b3873b0d-d50a-4ac9-a814-e50be0bf6602)

Reply:
I think this is awesome. Allowing users to configure multiple event types is going to help a ton. This gets my seal of approval. Thoughts @ThyMinimalDev @sean-brydon @ciaranha @itsberkaya?